File: app/src/main/java/com/jasonstudio/cookbook2/FavouriteRecipeModel/FavouriteRecipeViewModel.java
package com.jasonstudio.cookbook2.FavouriteRecipeModel;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;

import java.util.List;

import io.reactivex.Observable;

public class FavouriteRecipeViewModel extends AndroidViewModel {
    private FavouriteRecipeRepository repository;
    private Observable<List<FavouriteRecipe>> allList;

    public FavouriteRecipeViewModel(@NonNull Application application) {
        super(application);
        repository = new FavouriteRecipeRepository(application);
        allList = repository.getAllFR();
    }

    public void insert(FavouriteRecipe recipe){
        repository.insert(recipe);
    }
    public void delete(FavouriteRecipe recipe){
        repository.delete(recipe);
    }
    public void deleteAll(){
        repository.deleteAllFR();
    }

    public void deleteWithKey(int key) {
        repository.deleteWithKey(key);
    }

    public Observable<List<FavouriteRecipe>> getAll(){
        return allList;
    }


}

File: app/src/main/java/com/jasonstudio/cookbook2/FavouriteRecipeModel/FavouriteRecipeViewModel.kt
package com.jasonstudio.cookbook2.FavouriteRecipeModel

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.viewModelScope
import io.reactivex.Observable
import kotlinx.coroutines.launch

class FavouriteRecipeViewModel(application: Application) : AndroidViewModel(application) {
    private val repository: FavouriteRecipeRepository
    val all: Observable<List<FavouriteRecipe>>

    init {
        repository = FavouriteRecipeRepository(application)
        all = repository.allFR
    }

    fun insert(recipe: FavouriteRecipe?) {
        viewModelScope.launch {
            repository.insert(recipe)
        }
    }

    fun delete(recipe: FavouriteRecipe?) {
        viewModelScope.launch {
            repository.delete(recipe)
        }
    }

    fun deleteAll() {
        viewModelScope.launch {
            repository.deleteAllFR()
        }
    }

    fun deleteWithKey(key: Int) {
        viewModelScope.launch {
            repository.deleteWithKey(key)
        }
    }
}